src/mongocxx/database.cpp

```diff
@@ -157,5 +164,16 @@ collection database::collection(stdx::string_view name) const {
     return mongocxx::collection(*this, name);
 }
 
+const database::impl& database::get_impl() const {
+    if(!_impl) {
+        throw operation_exception{make_error_code(error_code::k_database_uninitialized)};
+    }
+    return *_impl;
+}
+
+database::impl* database::get_impl() {
+    return &(const_cast<database::impl&>(static_cast<const database&>(*this).get_impl()));
```

Comment:
can you put the implementation of this function in the non-const overload so you can avoid the need for const_cast?